Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -301,10 +301,9 @@ class PurchasesOrchestrator {
                   payment: SKMutablePayment,
                   package: Package?,
                   completion: @escaping PurchaseCompletedBlock) {
-        guard let productIdentifier = sk1Product.extractProductIdentifier(withPayment: payment) else {
-            Logger.error(Strings.purchase.could_not_purchase_product_id_not_found)
-            let errorMessage = "There was a problem purchasing the product: productIdentifier was nil"
-            completion(nil, nil, ErrorUtils.unknownError(message: errorMessage), false)
+        guard let productIdentifier = payment.extractProductIdentifier() else {
+            let message = Strings.purchase.could_not_purchase_product_id_not_found.description
+            completion(nil, nil, ErrorUtils.storeProblemError(withMessage: message), false)
```

Comment:
why are we ignoring the product identifier from the SKProduct now? 
The order should be: 
try getting the product id from the product, _then_ from the payment, and then if we still have nothing, raise an error

Reply:
That's explained in the PR description 😇

Reply:
Basically that was nice, the problem is when the transaction comes back, we only have that, and if the identifier there is `nil` we never called the callback. So the test I wrote (for example) hung forever.

So because we didn't have test coverage for that workaround we didn't realize that it was enough to begin the purchase, but then we can't complete it.

Reply:
I read the description but my brain didn't process it 🤦‍♂️ my apologies. 
That makes total sense. I keep thinking that there has to be something that we can do to improve this case, but ultimately I feel like the solution will be to start moving people to SK2 as it matures. 
In the meantime, I do think we should have an extensive comment here, since it won't be immediately obvious to folks maintaining this

Reply:
> but ultimately I feel like the solution will be to start moving people to SK2 as it matures.

Agreed.

> In the meantime, I do think we should have an extensive comment here, since it won't be immediately obvious to folks maintaining this

Good call, adding.